lib/posix/nanosleep.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2018 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <limits.h>
+#include <errno.h>
+/* required for struct timespec */
+#include <posix/time.h>
+
+#define NS_PER_SEC 1000000000ULL
+#define NS_PER_USEC 1000ULL
+
+/**
+ * @brief Suspend execution for nanosecond intervals.
+ *
+ * See IEEE 1003.1
+ */
+int nanosleep(const struct timespec *rqtp, struct timespec *rmtp)
+{
+	u64_t ns;
+	const bool write = (rmtp != NULL);
```

Comment:
Better would be something like `do_write` so that when grepping for `write` function we would not get this one.
Also () are superfluous and could be removed, but if you think it improves readability then I am fine with this too.